### DISCUSSION TITLE
Insert at position?

### DISCUSSION THREAD

**Discussion by privateOmega** [oozcitak/xmlbuilder2#108] [Q&A] [answered]
The only way I know to add child nodes to parent is via import, but is it possible to insert it as the first child or specifically at a position?

Appreciate the help.

NB: I am on xmlbuilder2@2.1.2

**Comment by privateOmega** [comment, marked as answer]
I figured out a crude way, but not sure if it has any other repurcussions that I don't know of, but my method at present is as follows.

```
const root = XMLFragment.root();
root.node.insertBefore(otherFragment.node, root.node.firstChild);
```

**Reply by oozcitak** [reply]
I should probably add `insertBefore ` and `insertAfter` functions but your workaround is perfectly fine.
